### PR TITLE
fix(issue-reports): Remove encoded solution

### PR DESCRIPTION
**Pull request by raisedadead**
#### Pre-Submission Checklist

- [x] Your pull request targets the `staging` branch of FreeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change

- [x] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.
- [ ] Closes currently open issue (replace XXXX with an issue no): Closes #XXXX

#### Description

This commit removes the encoded solution from the URI to make issue submission cleaner.

### CODE REVIEW

**Review by systimotic — COMMENTED** [review]
Should this be tied to #11261?

**Comment by raisedadead** [comment]
@systimotic nope, that's a different topic. It relates to how challenge solutions are encoded and evaluated in browsers. 

My PR addresses the GitHub Issue requests that are created from the Report Bug button, often campers reporting from the interface end up with issues that have huge URL in the description making it messy for everyone to understand. 

This is not required because the code from the editor is taken into the body anyway. 

My PR removes the same from the URL to the challenge. Hopefuly making it cleaner.

**Comment by systimotic** [comment]
Fair enough. My reasoning was that if the code is removed from the URL in general, we can send the full URL again.

I do quite like the way it is right now, but I could see why you would want to remove it, as it breaks often, which is really annoying.



On Tue, Nov 29, 2016 at 11:23 PM +0100, "Mrugesh Mohapatra" <notifications@github.com<mailto:notifications@github.com>> wrote:


@systimotic<https://github.com/systimotic> nope, that's a different topic. It relates to how challenge solutions are encoded and evaluated in browsers.

My PR addresses the GitHub Issue requests that are created from the Report Bug button, often campers reporting from the interface end up with issues that have huge URL in the description making it messy for everyone to understand.

This is not required because the code from the editor is taken into the body anyway.

My PR removes the same from the URL to the challenge. Hopefuly making it cleaner.

-

**Comment by Bouncey** [comment]
There is no encoded solution in the URI now with react.

This is the output for the issue template from `staging`
> `Challenge [say-hello-to-html-elements](http://localhost:3000/en/challenges/html5-and-css/say-hello-to-html-elements) has an issue.`

**Comment by raisedadead** [comment]
> There is no encoded solution in the URI now with react.

Agreed, but this does not count already constructed URLs in the Camper's profiles. Although really rare if someone would report after visiting a solution, yet this is a safety and the former logic of using `href` is not needed any longer.

my two cents.

File: client/commonFramework/report-issue.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ window.common = (function({ common = { init: [] } }) {
         'Challenge [',
         (common.challengeName || window.location.pathname),
         '](',
-        window.location.href,
```
Comment (Bouncey) [inline client/commonFramework/report-issue.js]: This file is no longer in use. 

Please edit the [new version of this file](https://github.com/FreeCodeCamp/FreeCodeCamp/blob/549cfc305f75d8c6036a3cde65c47b518607b18c/common/app/routes/challenges/redux/bug-saga.js). Line 53 is where the location is being sent with the issue :+1:

**Review by Bouncey — CHANGES_REQUESTED** [review]

**Comment by raisedadead** [comment]
Closing this request in favor of pushing this against `backup/master`. We are getting very frequent issues with URLs that aren't parsed by GitHub correctly.

**Pull request closed without merging**